python-sdk/src/uto/universal_transfer_operator.py

```diff
@@ -0,0 +1,76 @@
+from __future__ import annotations
+
+from typing import Any
+
+from airflow.models import BaseOperator
+from airflow.utils.context import Context
+from uto.datasets.base import UniversalDataset as Dataset
+from uto.utils import (
+    IngestorSupported,
+    check_if_connection_exists,
+    create_dataprovider,
+    create_transfer_integration,
+)
+
+from astro.constants import LoadExistStrategy
+
+
+class UniversalTransferOperator(BaseOperator):
+    """
+    Transfers all the data that could be read from the source Dataset into the destination Dataset. From a DAG author
+    standpoint, all transfers would be performed through the invocation of only the Universal Transfer Operator.
+
+    :param source_dataset: Source dataset to be transferred.
+    :param destination_dataset: Destination dataset to be transferred to.
+    :param extras: Extra arguments that may be passed for transfer operations.
+    :param use_optimized_transfer: Use use_optimized_transfer for data transfer if available on the destination.
+    :param optimization_params: kwargs to be used by method involved in optimized transfer flow.
+    :param ingestion_type: Use ingestion_type for methods involved in transfer using FiveTran.
+    :param ingestion_config: kwargs to be used by methods involved in transfer using FiveTran.
+    :param if_exists: Overwrite file if exists. Default False.
+
+    :return: returns the destination dataset
+    """
+
+    def __init__(
+        self,
+        *,
+        source_dataset: Dataset,
+        destination_dataset: Dataset,
+        extras: dict | None = {},
+        use_optimized_transfer: bool = True,
```

Comment:
`use_optimized_transfer` and `optimization_params` looks a bit confusing to me too, just for my understanding can you please elaborate more on it?